src/components/Preview.js

```diff
@@ -83,15 +88,18 @@ class Preview extends Component {
           {numFrames ? `${previewIdx + 1} / ${numFrames}` : '0 / 0'}
         </div>
         <div className="Preview-create">
-          {!!numFrames && (
-            <button
-              className="Preview-create-button"
-              onClick={this.handleGenerateGIF}
-              aria-label="generate gif"
-            >
-              Generate GIF
-            </button>
-          )}
+          {!!numFrames && this.props.gifData.length === 0 ? (
+            <GenerateGifForm
+              handleGenerateGIF={this.handleGenerateGIF}
```

Comment:
we're passing so much stuff in here, i'm wondering whether it would be better to just connect the form. would that simplify the preview container? what do y'all think?

Reply:
Done

Reply:
Yeah, this is a good call.